test/app.js

```diff
@@ -169,3 +169,49 @@ describe('generator-camel:app', function () {
     });
   });
 });
+
+describe('generator-camel:wsdl2rest', function () {
+
+  describe('Should properly scaffold with config for Spring and wsdl2rest', function () {
+
+    before(function () {
+      basicProps.name = 'MyAppMock';
+      basicProps.package = 'com.generator.mock';
+      basicProps.camelVersion = '2.18.2';
+      basicProps.camelDSL = 'spring';
+      var wsdlPath = path.join(__dirname, '../test/address.wsdl');
+      basicProps.wsdl = wsdlPath;
+      basicProps.outdirectory = 'src/main/java';
+
+      return helpers.run(path.join(__dirname, '../app'))
+        .inTmpDir(function (dir) {
+          var done = this.async(); // `this` is the RunContext object.
+          fs.copy(path.join(__dirname, '../templates'), dir, done);
+          basicProps.outdirectory = path.join(dir, 'src/main/java');
+        })
+        .withOptions({ wsdl2rest: true })
+        .withPrompts({ name: basicProps.name })
+        .withPrompts({ camelVersion: basicProps.camelVersion })
+        .withPrompts({ camelDSL: basicProps.camelDSL })
+        .withPrompts({ package: basicProps.package })
+        .withPrompts({ wsdl: basicProps.wsdl })
+        .withPrompts({ outdirectory: basicProps.outdirectory })
+        .toPromise().then(function(value) {
+          it('Should create the basic structure two ways', function () {
+            console.log('testing...');
+            assert.file('pom.xml');
+            assert.file('README.md');
+            assert.file('src/main/resources/META-INF/spring/camel-context.xml');
+            assert.file('src/main/resources/META-INF/spring/camel-context-rest.xml')
+          });
+        });
+    });
+
+    // it('Should create the basic structure two ways', function () {
```

Comment:
commented test:
is it because it is not working yet? if yes, is it possible to put some @Ignore with a reason why or a JIRA number to work on the issue.

Reply:
It's commented and moved up to the promise of the test before it because the wsdl2rest process has not finished by the time I'm looking for the artifacts it generates. I don't think this is actually running correctly the way I have it written and am still playing with things to see how to get this to test more effectively. Open to suggestions since I'm still trying to figure out this whole "promise" asynchronous way of doing things in JS.

Reply:
I have uncommented and moved it back because I don't believe it was being run.